State.py

```diff
@@ -166,6 +166,14 @@ def remove(self, item):
                 del self.prog_items[item.name]
 
 
+    def region_has_shortcuts(self, region_name, fallback_dungeon):
+        region = self.world.get_region(region_name)
+        dungeon_name = (region.dungeon and region.dungeon.name) or fallback_dungeon
```

Comment:
This could use `reverse_boss_map` instead of the `dungeon` field.

Reply:
`reverse_boss_map` just is {"New Boss Location Name": "Vanilla Boss Location Name"}  (or possibly the other way around) -- it doesn't have actual dungeon names at all.  It's intended for pedestal hints and things like showing the correct stone/medallion in Jabu-Jabu cutscenes.

This method's sole purpose, by the way, is King Dodongo and shortcuts logic (making water temple shortcuts potentially useful in this edge case).  No other boss room is affected by (or needs to be affected by) shortcuts.